test/Npgsql.PluginTests/NodaTimeTests.cs

```diff
@@ -364,5 +363,55 @@ protected override NpgsqlConnection OpenConnection(string? connectionString = nu
         }
 
         #endregion Support
+
+        [Test, Description("Makes sure noda time type handlers are forwarded to BCL type handlers.")]
+        public void BclTypeHandlerForwarding()
```

Comment:
I think this test should be redundant - the specific NodaTime tests should do all the testing. For example, test Interval above already checks that values can be read as a BCL TimeSpan - it just needs to add another parameter to test writing of TimeSpan as well.

Reply:
Since it's a workaround to support both type handlers at once let's add the provider specific types.

Reply:
You mean I should remove this additional test method and update the specific test methods accordingly?

Reply:
Yeah, that's my view.

There's also the question of if you want to support the provider-specific types (NpgsqlDate) in addition to the BCL types (DateTime). I definitely don't think this is mandatory, but if you want to you can again add testing in the existing tests.